components/admin_console/feature_discovery/feature_discovery.tsx

```diff
@@ -112,29 +117,46 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
                 defaultMessage='Start trial'
             />
         );
-        if (this.props.isCloud) {
+        if (this.props.isCloud && !this.props.isCloudFreeEnabled) {
             primaryMessage = (
                 <FormattedMessage
                     id='admin.ldap_feature_discovery_cloud.call_to_action.primary'
                     defaultMessage='Subscribe now'
                 />
             );
         }
+        const {
+            isCloud,
+            isCloudTrial,
+            isCloudFreeEnabled,
+            hadPrevCloudTrial,
+            isCloudFreePaidSubscription,
+        } = this.props;
+        const canRequestCloudFreeTrial = isCloud && isCloudFreeEnabled && !isCloudTrial && !hadPrevCloudTrial && !isCloudFreePaidSubscription;
+
+        const ctaTrialButton = canRequestCloudFreeTrial ? (
+            <CloudStartTrialButton
+                message={Utils.localizeMessage('menu.cloudFree.tryEnterpriseFor30Days', 'Try Enterprise free for 30 days')}
+                telemetryId={'start_cloud_trial_feature_discovery'}
```

Comment:
Is there any way to know what feature this is for? Maybe the title? Would be useful to have that in the telemetry

Reply:
makes sense, let me check!